code/modules/transport/pods/MainWeapon.dm

```diff
@@ -357,6 +357,123 @@
 		opencomputer(usr)
 		return
 
+/obj/item/shipcomponent/mainweapon/syndicate_purge_system
+	name = "Syndicate Purge System"
+	desc = "An unfinished pod weapon, the blueprints for which have been plundered from a raid on a now-destroyed Syndicate base. Requires a unique power source to function."
+	current_projectile = new/datum/projectile/laser/drill/cutter
+	firerate = 100
+	var/increment
+	var/pod_is_large = false
+	var/core_inserted = false
+	var/image/purge
+	icon = 'icons/misc/retribution/SWORD_loot.dmi'
+	icon_state= "SPS_empty"
+
+	Fire(var/mob/user,var/shot_dir_override = -1)
+		if(!core_inserted)
+			boutput(ship.pilot, "<span class='alert'><B>The weapon requires a unique power source to function!</B></span>")
+			return
+		if(isfiring) return
+		isfiring = 1
+		playsound(src.loc, "sound/weapons/heavyioncharge.ogg", 75, 1)
+		logTheThing("combat", usr, null, "driving [ship.name] fires [src.name] from [log_loc(ship)].")
+		if(ship.capacity != 1 && !istype(/obj/machinery/vehicle/miniputt, ship) && !istype(/obj/machinery/vehicle/recon, ship) && !istype(/obj/machinery/vehicle/cargo, ship))
+			pod_is_large = true
+			purge = image('icons/misc/retribution/320x320.dmi', "SPS_o_large", "layer" = EFFECTS_LAYER_4)
+			purge.pixel_x -= 128
+			purge.pixel_y -= 128
+		else
+			pod_is_large = false
+			purge = image('icons/misc/retribution/320x320.dmi', "SPS_o_small", "layer" = EFFECTS_LAYER_4)
+			purge.pixel_x -= 144
+			purge.pixel_y -= 144
+		purge.plane = PLANE_SELFILLUM
+		ship.UpdateOverlays(purge, "purge")
+
+		spawn(12)
+			var/destruction_point_x
+			var/destruction_point_y
+			spawn(8)
+				ship.UpdateOverlays(null, "purge")
+			playsound(ship.loc, 'sound/weapons/laserultra.ogg', 100, 1)
+			switch (ship.dir)
+				if (1)	//N
+					for (increment=1; increment<=4; increment++)
```

Comment:
the byond for loop syntax is more efficient if I'm not mistaken

https://hackmd.io/@ZeWaka/gooncode#for-in-to-loops

Reply:
Huh, didn't know that, thanks.

Reply:
Done.